flink-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBaseTest.java

```diff
@@ -420,4 +446,15 @@ void enqueueCompletableFuture(CompletableFuture<ResultSet> completableFuture) {
 			throw new InvalidQueryException("For test purposes");
 		}
 	}
+
+	private static class SendErrorTestCassandraSink extends TestCassandraSink {
+		SendErrorTestCassandraSink(CassandraSinkBaseConfig config) {
+			super(config, new NoOpCassandraFailureHandler());
+		}
+
+		@Override
+		public ListenableFuture<ResultSet> send(String value) {
+			throw new Error("For test purposes");
```

Comment:
if we want to catch `Throwable`, could we also throw `Throwable`? otherwise we are testing that it catches only `Error`.

Reply:
Unfortunately I don't think we can do this, since we would then have to add `Throwable` to the signature of `send()` in the base class.

Reply:
true, we do not want this change